### PR TITLE
[RFR] Preload references in refrenced_list entries

### LINKED ISSUE #637: reference field in referenced_list?
Is it possible to have a reference field within a referenced list? eg:

```javascript
nga.field('accounts', 'referenced_list') // display list of related accounts
    .targetEntity(accounts) 
    .targetReferenceField('voip_id')
    .targetFields([
        nga.field('id').isDetailLink(true),
        nga.field('date_created'),
        nga.field('user_profile_id'), // <-- this works fine
        nga.field('user_profile_id', 'reference') // <-- it doesn't call the api request for the profile entity
            .label('User Profile')
            .validation({required: true})
            .targetEntity(profiles)
            .targetField(nga.field('username')),
    ])
```

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] Implement the feature
- [x] Refactor the routing for more readbility
- [x] make tests pass

Closes #637, #383

### CODE REVIEW

**Comment by fzaninotto** [comment]
Switching to RFR

File: src/javascripts/ng-admin/Crud/column/maReferenceColumn.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        link: {
+            pre: function(scope) {
+                const value = scope.value();
+                scope.field = scope.field();
+                scope.targetEntity = scope.field.targetEntity();
+                scope.targetField = scope.field.targetField();
+                const identifierName = scope.targetEntity.identifier().name()
+                scope.referencedEntry = scope.datastore()
+                    .getEntries(scope.targetEntity.uniqueId + '_values')
+                    .filter(entry => entry.values[identifierName] == value)
```
Comment (jpetitcolas) [inline src/javascripts/ng-admin/Crud/column/maReferenceColumn.js]: `===`

File: src/javascripts/ng-admin/Crud/column/maReferencedListColumn.js [hunk 2]
```diff
@@ -17,8 +17,9 @@ function maReferencedListColumn(NgAdminConfiguration) {
 <ma-datagrid name="{{ field.datagridName() }}"
     entries="::entries"
     fields="::field.targetFields()"
-    list-actions="::field.listActions()" 
-    entity="::entity">
+    list-actions="::field.listActions()"
+    entity="::entity"
+    datastore="::datastore()">
```
Comment (jpetitcolas) [inline src/javascripts/ng-admin/Crud/column/maReferencedListColumn.js]: Can't we retrieve `entries` from the datastore now?

Comment (jpetitcolas) [inline src/javascripts/ng-admin/Crud/column/maReferencedListColumn.js]: Got my response IRL: we can't remove entries, as this is the only way to keep entry ids.

**Comment by jpetitcolas** [comment]
:+1:

File: src/javascripts/ng-admin/Crud/column/maReferenceColumn.js (hunk 1 above)
Comment (fzaninotto) [inline src/javascripts/ng-admin/Crud/column/maReferenceColumn.js]: Assuming that the APIs we fetch are consistent (i.e. that the type of `comment.post_id` is the same as the type of `post.id`) is pretty brave, but I think we'd rather be tolerant to ease compatibility. So e.g. when `comment.post_id` is `'1'` and `post.id` is `1`, the current code works.

File: src/javascripts/ng-admin/Crud/column/maReferencedListColumn.js (hunk 2 above)
Comment (fzaninotto) [inline src/javascripts/ng-admin/Crud/column/maReferencedListColumn.js]: Yep, but we'd need to pass an array of ideas and a type to let the datagrid retrieve the entries from the datastore... not ideal.

**Pull request merged**